### ISSUE TITLE
False positive if a Pydantic model is instantiated with a kwargs dict

### ISSUE THREAD

**Issue by cbornet** [astral-sh/ty#4072]
### Summary

Instantiating a Pydantic model with a kwargs dict results in an error if the dict types don't match the model.

```python
from pydantic import BaseModel

class MyClass(BaseModel):
    name: str | None = None

params: dict[str, Any] = {'foo': 1}
MyClass(**params)  # ok

MyClass(**{'foo': 1})  # error[invalid-argument-type]: Argument is incorrect. 
       #^^^^^^^^^^^^ Expected `LaxStr | None`, found `int`

MyClass(**{'foo': 'bar'})   # ok...

MyClass(foo=1)  # warning[pydantic-discarded-extra-argument]: Extra argument `foo` is discarded by Pydantic
```

Expected: there should be no error for `MyClass(**{'foo': 1})`. There could be at most a `pydantic-discarded-extra-argument` warning if it's possible to detect the key names.

### Version

0.0.62

**Comment by sharkdp** [comment]
Thank you for reporting this.

The problem that you discovered here goes beyond our Pydantic support (see https://github.com/astral-sh/ty/issues/3792). I'm leaving this open, but I think we should discuss the underlying issue in that ticket.

**Comment by cbornet** [comment]
Indeed, it seems related.
Though note that with mypy, this case is handled by the `pydantic.mypy` plugin:
* Without plugin, `error: Argument 1 to "MyClass" has incompatible type "**dict[str, int]"; expected "str | None"  [arg-type]`
* With plugin, no error.

It would be nice to handle it the same way as the `pydantic.mypy` plugin for easy transition.
